cpp/dolfinx/io/ADIOS2Writers.h

```diff
@@ -867,6 +867,330 @@ void vtx_write_mesh_from_space(adios2::IO& io, adios2::Engine& engine,
+  ///
+  /// This format supports arbitrary degree meshes.
+  ///
+  /// @param[in] comm The MPI communicator to open the file on
+  /// @param[in] filename Name of output file
+  /// @param[in] mesh The mesh
+  /// @note This format support arbitrary degree meshes
+  ADIOS2Checkpointer(MPI_Comm comm, const std::filesystem::path& filename,
+                     std::shared_ptr<const mesh::Mesh<T>> mesh)
+      : _adios(std::make_unique<adios2::ADIOS>(comm)),
+        _io(std::make_unique<adios2::IO>(
+            _adios->DeclareIO("ADIOS2 checkpoint writer"))),
+        _engine(std::make_unique<adios2::Engine>(
+            _io->Open(filename, adios2::Mode::Write))),
+        _mesh(mesh), _u({}), _comm(comm)
+  {
+    // Define time independent attributes
+    auto topology = mesh->topology();
+    assert(topology);
+    if (topology->cell_types().size() > 1)
+      throw std::runtime_error("Multiple cell types not supported");
+    impl_adios2::define_attribute<int>(*this->_io, "CellType",
+                                       int(topology->cell_types()[0]));
+
+    const mesh::Geometry<T>& geometry = mesh->geometry();
+    if (geometry.cmaps().size() > 1)
+      throw std::runtime_error("Multiple coordinate maps not supported");
+    impl_adios2::define_attribute<int>(*this->_io, "Degree",
+                                       geometry.cmaps()[0].degree());
+    impl_adios2::define_attribute<int>(*this->_io, "LagrangeVariant",
+                                       int(geometry.cmaps()[0].variant()));
+  }
+
+  /// @brief Create a ADIOS2 checkpointer for reading a mesh from file
+  ///
+  /// This format supports arbitrary degree meshes.
+  ///
+  /// @param[in] comm The MPI communicator to open the file on
+  /// @param[in] filename Name of input file
+  /// @note This format support arbitrary degree meshes
+  ADIOS2Checkpointer(MPI_Comm comm, const std::filesystem::path& filename)
+      : _adios(std::make_unique<adios2::ADIOS>(comm)),
+        _io(std::make_unique<adios2::IO>(
+            _adios->DeclareIO("ADIOS2 checkpoint reader"))),
+        _engine(std::make_unique<adios2::Engine>(
+            _io->Open(filename, adios2::Mode::Read))),
+        _comm(comm)
+  {
+    // Do  nothing
+  }
+
+  /// @brief  Write data with a given time
+  /// @param[in] t The time step
+  void write(double t)
+  {
+    assert(this->_io);
+    assert(this->_engine);
+    adios2::Variable<double> var_step
+        = impl_adios2::define_variable<double>(*this->_io, "step");
+
+    this->_engine->BeginStep();
+    this->_engine->template Put<double>(var_step, t);
+
+    impl_checkpoint::write_mesh(*this->_io, *this->_engine, *this->_mesh);
+    // If we have no functions write the mesh to file
+    // if (this->_u.empty())
+
+    this->_engine->EndStep();
+  }
+
+  /// @brief  Read mesh from a given time-step
+  /// @param[in] t The time step
+  /// @param[in] mode The ghost mode
+  mesh::Mesh<T> read_mesh(double t, mesh::GhostMode mode)
```

Comment:
What format does it read?

Reply:
`.bp` files created with the `ADIOS2Checkpointer`